src/librustc/hir/mod.rs

```diff
@@ -340,12 +341,12 @@ impl PathSegment {
     }
 }
 
+pub type GenericPathParam = GenericParameter<Lifetime, P<Ty>>;
```

Comment:
IMO this should be called just `PathParam` (and `PathParameters` below just `PathParams`). Same for `parameters` vs `params`, but avoid changing too much existing code unless you want a separate PR.